### PR TITLE
CNF-14656: CNF-14653: ClusterRequest to ProvisioningRequest and make it cluster scoped

### PR DESCRIPTION
**Pull request by irinamihai**
- Rename `ClusterRequest` to `ProvisioningRequest`
- Make `ProvisioningRequest` cluster coped

### CODE REVIEW

**Comment by sudomakeinstall2** [comment]
Now that ProvisioningRequest is cluster-scoped, shouldn't we have templateNamespace field?

```
spec:
	templateRef:
	    name: clustertemplate-a                 
	    version: v1
	    namespace: cluster-templates-a-4.16      
```

**Comment by irinamihai** [comment]
> Now that ProvisioningRequest is cluster-scoped, shouldn't we have templateNamespace field?
> 
> ```
> spec:
> 	templateRef:
> 	    name: clustertemplate-a                 
> 	    version: v1
> 	    namespace: cluster-templates-a-4.16      
> ```

@sudomakeinstall2 , no, unfortunately. The current direction of the ORAN provisioning spec is to not include the namespace of the ClusterTemplate. The provisioning controller will choose the first valid ClusterTemplate with the specified name/version. The clusterTemplate controller will make sure to only allow unique ClusterTemplate names across the cluster. See @edcdavid 's latest [PR](https://github.com/openshift-kni/oran-o2ims/pull/219).

File: internal/controllers/provisioningrequest_controller.go [hunk 1]
```diff
@@ -83,9 +84,8 @@ type deleteOrUpdateEvent interface {
 }
 
 const (
-	clusterRequestFinalizer         = "clusterrequest.oran.openshift.io/finalizer"
-	clusterRequestNameLabel         = "clusterrequest.oran.openshift.io/name"
-	clusterRequestNamespaceLabel    = "clusterrequest.oran.openshift.io/namespace"
+	provisioningRequestFinalizer    = "provisioningrequest.oran.openshift.io/finalizer"
```
Comment (Missxiaoguo) [inline internal/controllers/provisioningrequest_controller.go:87]: Should these be changed to reflect with the new API group?

File: internal/controllers/provisioningrequest_controller.go [hunk 2]
```diff
@@ -1291,22 +1290,22 @@ func (t *clusterRequestReconcilerTask) createOrUpdateClusterResources(
 
 // createExtraManifestsConfigMap copies the extra-manifests ConfigMaps from the
 // cluster template namespace to the clusterInstance namespace.
-func (t *clusterRequestReconcilerTask) createExtraManifestsConfigMap(
+func (t *provisioningRequestReconcilerTask) createExtraManifestsConfigMap(
 	ctx context.Context, clusterInstance *siteconfig.ClusterInstance) error {
 	for _, extraManifestsRef := range clusterInstance.Spec.ExtraManifestsRefs {
 		// Make sure the extra-manifests ConfigMap exists in the clusterTemplate namespace.
-		// The clusterRequest namespace is the same as the clusterTemplate namespace.
+		// The ProvisioningRequest namespace is the same as the clusterTemplate namespace.
```
Comment (Missxiaoguo) [inline internal/controllers/provisioningrequest_controller.go:1297]: I think this comment is obslete now.

File: internal/controllers/provisioningrequest_controller.go [hunk 3]
```diff
@@ -1336,11 +1335,11 @@ func (t *clusterRequestReconcilerTask) createPullSecret(
 	// clusterName
 
 	// Check the pull secret already exists in the clusterTemplate namespace.
-	// The clusterRequest namespace is the same as the clusterTemplate namespace.
+	// The ProvisioningRequest namespace is the same as the clusterTemplate namespace.
```
Comment (Missxiaoguo) [inline internal/controllers/provisioningrequest_controller.go:1338]: obselete comment

**Comment by jc-rh** [comment]
> The provisioning controller will choose the first valid ClusterTemplate with the specified name/version.

Can you clarify "first"? We plan to sort them by namespace or creation time or some combination of different things?

File: bundle/manifests/oran-o2ims.clusterserviceversion.yaml [hunk 4]
```diff
@@ -283,6 +751,52 @@ spec:
         x-descriptors:
         - urn:alm:descriptor:com.tectonic.ui:text
       version: v1alpha1
+    - description: ProvisioningRequest is the Schema for the provisioningrequests
+        API
+      displayName: ORAN O2IMS Cluster Request
```
Comment (bartwensley) [inline bundle/manifests/oran-o2ims.clusterserviceversion.yaml:756]: Cluster -> Provisioning

File: config/manifests/bases/oran-o2ims.clusterserviceversion.yaml [hunk 5]
```diff
@@ -257,6 +212,52 @@ spec:
         x-descriptors:
         - urn:alm:descriptor:com.tectonic.ui:text
       version: v1alpha1
+    - description: ProvisioningRequest is the Schema for the provisioningrequests
+        API
+      displayName: ORAN O2IMS Cluster Request
```
Comment (bartwensley) [inline config/manifests/bases/oran-o2ims.clusterserviceversion.yaml:217]: Cluster -> Provisioning

Comment (irinamihai) [reply config/manifests/bases/oran-o2ims.clusterserviceversion.yaml:217]: Right, there are a bunch of occurrences with the space included. I'll fix them asap. Thanks!

Comment (irinamihai) [reply config/manifests/bases/oran-o2ims.clusterserviceversion.yaml]: Ok, done.

File: api/hardwaremanagement/v1alpha1/node.go [hunk 6]
```diff
@@ -65,7 +65,7 @@ type NodeStatus struct {
 //
 // +kubebuilder:object:root=true
 // +kubebuilder:subresource:status
-// +operator-sdk:csv:customresourcedefinitions:displayName="ORAN O2IMS Cluster Request",resources={{Namespace, v1}}
+// +operator-sdk:csv:customresourcedefinitions:displayName="ORAN O2IMS Provisioning Request",resources={{Namespace, v1}}
```
Comment (bartwensley) [inline api/hardwaremanagement/v1alpha1/node.go:68]: I know this was wrong before, but I think the displayName should just be "Node" here.

File: api/hardwaremanagement/v1alpha1/node_pools.go [hunk 7]
```diff
@@ -84,7 +84,7 @@ type NodePoolStatus struct {
 // +kubebuilder:resource:shortName=np
 // +kubebuilder:object:root=true
 // +kubebuilder:subresource:status
-// +operator-sdk:csv:customresourcedefinitions:displayName="ORAN O2IMS Cluster Request",resources={{Namespace, v1}}
+// +operator-sdk:csv:customresourcedefinitions:displayName="ORAN O2IMS Provisioning Request",resources={{Namespace, v1}}
```
Comment (bartwensley) [inline api/hardwaremanagement/v1alpha1/node_pools.go:87]: I know this was wrong before, but I think the displayName should just be "NodePool" here.

File: bundle/manifests/oran-o2ims.clusterserviceversion.yaml [hunk 8]
```diff
@@ -231,7 +707,7 @@ spec:
         path: usedServerConfig
       version: v1alpha1
     - description: NodePool is the schema for an allocation request of nodes
-      displayName: ORAN O2IMS Cluster Request
+      displayName: ORAN O2IMS Provisioning Request
```
Comment (bartwensley) [inline bundle/manifests/oran-o2ims.clusterserviceversion.yaml:710]: displayName should be NodePool

File: bundle/manifests/oran-o2ims.clusterserviceversion.yaml [hunk 9]
```diff
@@ -259,7 +735,7 @@ spec:
         - urn:alm:descriptor:com.tectonic.ui:text
       version: v1alpha1
     - description: Node is the schema for an allocated node
-      displayName: ORAN O2IMS Cluster Request
+      displayName: ORAN O2IMS Provisioning Request
```
Comment (bartwensley) [inline bundle/manifests/oran-o2ims.clusterserviceversion.yaml:738]: displayName should be Node

File: api/hardwaremanagement/v1alpha1/node.go (hunk 6 above)
Comment (irinamihai) [reply api/hardwaremanagement/v1alpha1/node.go]: Done

File: api/hardwaremanagement/v1alpha1/node_pools.go (hunk 7 above)
Comment (irinamihai) [reply api/hardwaremanagement/v1alpha1/node_pools.go]: Done

File: bundle/manifests/oran-o2ims.clusterserviceversion.yaml (hunk 8 above)
Comment (irinamihai) [reply bundle/manifests/oran-o2ims.clusterserviceversion.yaml]: Done

File: bundle/manifests/oran-o2ims.clusterserviceversion.yaml (hunk 9 above)
Comment (irinamihai) [reply bundle/manifests/oran-o2ims.clusterserviceversion.yaml]: Done

**Pull request merged**